membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -269,23 +303,72 @@ class AccountController(commonActions: CommonActions, override val controllerCom
-      upToDatePaymentDetails <- ListEither.liftList(getUpToDatePaymentDetailsFromStripe(contactAndSubscription.subscription.accountId, paymentDetails).map(\/.right).recover { case x => \/.left(s"error getting up-to-date card details for payment method of account: ${contactAndSubscription.subscription.accountId}. Reason: $x") })
-      accountSummary <- ListEither.liftList(tp.zuoraRestService.getAccount(contactAndSubscription.subscription.accountId).recover { case x => \/.left(s"error receiving account summary for subscription: ${contactAndSubscription.subscription.name} with account id ${contactAndSubscription.subscription.accountId}. Reason: $x") })
-      effectiveCancellationDate <- ListEither.liftList(tp.zuoraRestService.getCancellationEffectiveDate(contactAndSubscription.subscription.name).recover { case x => \/.left(s"Failed to fetch effective cancellation date: ${contactAndSubscription.subscription.name} with account id ${contactAndSubscription.subscription.accountId}. Reason: $x") })
-      stripeService = accountSummary.billToContact.country.map(RegionalStripeGateways.getGatewayForCountry).flatMap(tp.stripeServicesByPaymentGateway.get).getOrElse(tp.ukStripeService)
+      paymentDetails <- ListEither.liftList(
+        getPaymentDetailsForSub(maybeUserId, freeOrPaidSub, tp.paymentService).map(\/.right).recover {
+          case x => \/.left(s"error retrieving payment details for subscription: freeOrPaidSub.name. Reason: $x")
+        }
+      )
+      upToDatePaymentDetails <- ListEither.liftList(
+        getUpToDatePaymentDetailsFromStripe(contactAndSubscription.subscription.accountId, paymentDetails).map(\/.right).recover {
+          case x => \/.left(s"error getting up-to-date card details for payment method of account: " +
+            s"${contactAndSubscription.subscription.accountId}. Reason: $x")
+        })
+      accountSummary <- ListEither.liftList(
+        tp.zuoraRestService.getAccount(contactAndSubscription.subscription.accountId).recover {
+          case x => \/.left(s"error receiving account summary for subscription: ${contactAndSubscription.subscription.name} " +
+            s"with account id ${contactAndSubscription.subscription.accountId}. Reason: $x")
+        })
+      effectiveCancellationDate <- ListEither.liftList(
+        tp.zuoraRestService.getCancellationEffectiveDate(contactAndSubscription.subscription.name).recover {
+          case x => \/.left(s"Failed to fetch effective cancellation date: ${contactAndSubscription.subscription.name} " +
+            s"with account id ${contactAndSubscription.subscription.accountId}. Reason: $x")
+        })
+      stripeService = accountSummary.billToContact.country.map(RegionalStripeGateways.getGatewayForCountry)
+        .flatMap(tp.stripeServicesByPaymentGateway.get)
+        .getOrElse(tp.ukStripeService)
```

Comment:
this is very hard to read! I can see you have tidied it up a little which is great.  monad transfomers plus inline recover functions really add a lot of clutter.  I assume there are no substantial changes here.

Reply:
No only formatting changes